### ISSUE TITLE
Issue when using in GraalVM

### ISSUE THREAD

**Issue by etspaceman** [vlovgr/ciris#452]
Hey there - 

I recently had to drop my use of Ciris in one of my projects due to the inability to use it properly in GraalVM.  I've attached the full stack trace. Switching to Pureconfig solved this for me. I was also able to solve this using `ConfigValue.default`. Seemed like the issue was being thrown in the `flatMap` command here: https://github.com/vlovgr/ciris/blob/master/modules/core/src/main/scala/ciris/ConfigValue.scala#L155

[message.txt](https://github.com/vlovgr/ciris/files/6298369/message.txt)

https://github.com/etspaceman/kinesis-mock/tree/4acc3ee60a6dda9ea85629c2551a25faa9985ac6

You can reproduce this by running `sbt packageAndBuildDockerImage`.

**Comment by vlovgr** [comment]
I see you're using v1.2.1, so [this line](https://github.com/vlovgr/ciris/blob/66a26dd3728aefb0ca6ba501d0206121ce47b2d8/modules/core/src/main/scala/ciris/ConfigValue.scala#L155) is the actual failing one.
From what I can tell, this is a GraalVM issue, see [this comment](https://github.com/oracle/graal/issues/2761#issuecomment-785006081).

**Comment by etspaceman** [comment]
Yeah I saw that in my research. I was unable to find an actual issue opened due to this.

**Comment by etspaceman** [comment]
Created https://github.com/oracle/graal/issues/3341
